modules/programs/thunderbird.nix

```diff
@@ -0,0 +1,267 @@
+    programs.thunderbird = {
+      enable = mkEnableOption "Thunderbird";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.thunderbird;
+        defaultText = literalExpression "pkgs.thunderbird";
+        example = literalExpression "pkgs.thunderbird-91";
+        description = "The Thunderbird package to use.";
+      };
+
+      profiles = mkOption {
+        type = with types;
+          attrsOf (submodule ({ config, name, ... }: {
+            options = {
+              name = mkOption {
+                type = types.str;
+                default = name;
+                readOnly = true;
+                description = "This profile's name.";
+              };
+
+              settings = mkOption {
+                type = with types; attrsOf (oneOf [ bool int str ]);
+                default = { };
+                example = literalExpression ''
+                  {
+                    "mail.spellcheck.inline" = false;
+                  }
+                '';
+                description = ''
+                  Preferences to add to this profile's
+                  <filename>user.js</filename>.
+                '';
+              };
+
+              isDefault = mkOption {
+                type = types.bool;
+                default = false;
+                example = true;
+                description = ''
+                  Whether this is a default profile. There must be exactly one
+                  default profile.
+                '';
+              };
+            };
+          }));
+      };
+
+      settings = mkOption {
+        type = with types; attrsOf (oneOf [ bool int str ]);
+        default = { };
+        example = literalExpression ''
+          {
+            "general.useragent.override" = "";
+            "privacy.donottrackheader.enabled" = true;
+          }
+        '';
+        description = ''
+          Attribute set of Thunderbird preferences to be added to
+          all profiles.
+        '';
+      };
+
+      withExternalGnupg = mkOption {
+        type = types.bool;
+        default = false;
+        example = true;
+        description = "Allow using external GPG keys with GPGME.";
+      };
+    };
+
+    accounts.email.accounts = mkOption {
+      type = with types;
+        attrsOf (submodule {
+          options.thunderbird = {
+            enable =
+              mkEnableOption "the Thunderbird mail client for this account";
+
+            profiles = mkOption {
+              type = with types; listOf str;
+              default = [ ];
+              example = literalExpression ''
+                [ "profile1" "profile2" ]
+              '';
+              description = ''
+                List of Thunderbird profiles for which this account should be
+                enabled. If this list is empty (the default), this account will
+                be enabled for all declared profiles.
+              '';
+            };
+
+            settings = mkOption {
```

Comment:
Without understanding use case the whole option seems a bit complex.

Id is not stable here either maybe.

Reply:
The idea is that per-account options might need to have access to the account id. I thought using a function was cleaner than having to copy the id from a previously generated `user.js` file.

Reply:
Ok so it's just a string template with the names, I get it now, I guess that's fine.